### PR TITLE
Reformatting testharness extra stuff for pending concatenation onto bootstrap.js

### PR DESCRIPTION
**Pull request by mithro**

### CODE REVIEW

**Comment by mithro** [comment]
I don't think diff is going to give you anything useful here?

File: test/testharness_extensions.js [hunk 1]
```diff
@@ -25,345 +25,332 @@
...
+        callback.call(thisObj, array[i], i, array);
+      }
     }
+  }
 
-    /* ********************************************************************* */
+  /* ********************************************************************* */
 
-    var pageerror_test = async_test("Page contains no errors");
+  var pageerror_test = async_test("Page contains no errors");
```
Comment (dstoc) [inline test/testharness_extensions.js]: Prefer single quotes.

File: test/testharness_extensions.js [hunk 2]
```diff
@@ -25,345 +25,332 @@
...
-                callback.call(thisObj, array[i], i, array);
-            }
-        }
+  /**
+   * These functions come from testharness.js but can't be access because
+   * testharness uses an anonymous function to hide them.
+   **************************************************************************
+   */
+  function forEach (array, callback, thisObj) {
+    for (var i=0; i<array.length; i++) {
```
Comment (dstoc) [inline test/testharness_extensions.js]: Spaces around binary operators

File: test/testharness_extensions.js [hunk 3]
```diff
@@ -25,345 +25,332 @@
...
 
-    var pageerror_test = async_test("Page contains no errors");
+  var pageerror_test = async_test("Page contains no errors");
 
-    function pageerror_onerror_callback(evt) {
-        var msg = "Error in " + evt.filename + "\n" +
-                  "Line " + evt.lineno + ": " + evt.message + "\n";
+  function pageerror_onerror_callback(evt) {
+    var msg = "Error in " + evt.filename + "\n" +
+          "Line " + evt.lineno + ": " + evt.message + "\n";
```
Comment (dstoc) [inline test/testharness_extensions.js]: This only needs to be indented 4 spaces

File: test/testharness_extensions.js [hunk 4]
```diff
@@ -25,345 +25,332 @@
...
+  };
+  addEventListener('error', pageerror_onerror_callback);
 
-    var pageerror_tests;
-    function pageerror_othertests_finished(test, harness) {
-        if (harness == null && pageerror_tests == null)
-            return;
+  var pageerror_tests;
+  function pageerror_othertests_finished(test, harness) {
+    if (harness == null && pageerror_tests == null)
```
Comment (dstoc) [inline test/testharness_extensions.js]: Use braces even when the block only contains a single statement.

File: test/testharness_extensions.js [hunk 5]
```diff
@@ -25,345 +25,332 @@
...
+    if (pageerror_tests.all_loaded && pageerror_tests.num_pending == 1) {
+      pageerror_test.done();
+    }
+  }
+  add_result_callback(pageerror_othertests_finished);
+  addEventListener('load', pageerror_othertests_finished);
+
+  /* ********************************************************************* */
+  var svg_properties = {
+    'cx': 1,
```
Comment (dstoc) [inline test/testharness_extensions.js]: No need to quote 'cx', 'width'...

File: test/testharness_extensions.js [hunk 6]
```diff
@@ -25,345 +25,332 @@
...
+    'width': 1,
+    'x': 1,
+    'y': 1,
+  };
+
+  var is_svg_attrib = function(property, target) {
+    return target.namespaceURI == 'http://www.w3.org/2000/svg' &&
+      property in svg_properties;
```
Comment (dstoc) [inline test/testharness_extensions.js]: When wrapping, always indent at least 4 spaces.

File: test/testharness_extensions.js [hunk 7]
```diff
@@ -25,345 +25,332 @@
...
+  };
+
+  var svg_namespace_uri = 'http://www.w3.org/2000/svg';
+
+  var features = (
+    function () {
```
Comment (dstoc) [inline test/testharness_extensions.js]: Put function on the previous line and remove indentation.

Comment (dstoc) [inline test/testharness_extensions.js]: No space between function and ()

File: test/testharness_extensions.js [hunk 8]
```diff
@@ -25,345 +25,332 @@
...
+   *
+   * @param {string|Element|Array.<Element>} objects Either;
+   *    * A query selector to use to get DOM nodes,
+   *    * A DOM node.
+   *    * A list of DOM nodes.
+   * @param {Array.<Object.<string, string>>|Object.<string, string>} style
+   *   See _assert_style_get for information.
+   */
+  function assert_styles(objects, style, description) {
+    switch(typeof objects) {
```
Comment (dstoc) [inline test/testharness_extensions.js]: indent switch as:

```
switch (foo) {
  case bar:
    ...;
    break;
  case ...
```

File: test/testharness_extensions.js [hunk 9]
```diff
@@ -25,345 +25,332 @@
...
-                throw new Error("Expected Array, NodeList or Element but got " + objects);
-            }
-            break;
-        }
-    }
-
-    window.assert_styles = assert_styles;
+  window.assert_styles = assert_styles;
 })();
 // vim: set expandtab shiftwidth=4 tabstop=4:
```
Comment (dstoc) [inline test/testharness_extensions.js]: Shouldn't need this now

**Comment by dstoc** [comment]
Most of those issues are repeated multiple times, I just commented on the first instances.

File: test/testharness_extensions.js (hunk 1 above)
Comment (mithro) [inline test/testharness_extensions.js]: Done everywhere.

File: test/testharness_extensions.js (hunk 3 above)
Comment (mithro) [inline test/testharness_extensions.js]: Done.

File: test/testharness_extensions.js (hunk 4 above)
Comment (mithro) [inline test/testharness_extensions.js]: Done everywhere.

File: test/testharness_extensions.js (hunk 5 above)
Comment (mithro) [inline test/testharness_extensions.js]: Done.

File: test/testharness_extensions.js (hunk 7 above)
Comment (mithro) [inline test/testharness_extensions.js]: Both done.

File: test/testharness_extensions.js (hunk 8 above)
Comment (mithro) [inline test/testharness_extensions.js]: Done everywhere.

File: test/testharness_extensions.js (hunk 9 above)
Comment (mithro) [inline test/testharness_extensions.js]: Removed.

File: test/testharness_extensions.js (hunk 6 above)
Comment (mithro) [inline test/testharness_extensions.js]: Done.

**Comment by mithro** [comment]
I've tried to catch all of them, please point out if I missed anything. Please take another look.

Thanks for the detailed review - no idea how you were able to deal with that massive diff.

**Comment by dstoc** [comment]
> Thanks for the detailed review - no idea how you were able to deal with that massive diff.

I opened the full file beside the diff :)

**Comment by mithro** [comment]
Hopefully fixed all the issues you pointed out (and found a couple of others via regex). 

Anything more  I've missed?

**Comment by dstoc** [comment]
lgtm

**Pull request merged**